app/assets/javascripts/discourse/widgets/user-menu.js.es6

```diff
@@ -194,7 +194,23 @@ export default createWidget("user-menu", {
     });
   },
 
-  clickOutside() {
-    this.sendWidgetAction("toggleUserMenu");
+  clickOutside(e) {
+    const $centeredElement = $(document.elementFromPoint(e.clientX, e.clientY));
+    if (
+      $centeredElement.parents(".panel").length &&
+      !$centeredElement.hasClass("header-cloak")
+    ) {
+      this.sendWidgetAction("toggleUserMenu");
+    } else {
+      const $window = $(window);
+      const windowWidth = parseInt($window.width());
```

Comment:
It's good practice to always use the radix, even if it should never be an issue here.

```suggestion
      const windowWidth = parseInt($window.width(), 10);
```